src/main/java/vimification/model/Model.java

```diff
@@ -34,6 +38,30 @@ public interface Model {
      */
     void setGuiSettings(GuiSettings guiSettings);
 
+    //=========== Task oriented methods (NEW CODE) =============================================================
+
+    /**
+     * Adds the given new task.
+     */
+    void addTask(Task newTask);
+
+    /**
+     * Deletes the given task.
+     * The task must exist in the task list.
+     */
+    void deleteTask(Task target);
+
+    /**
+     * Updates the filter of the filtered task list to filter by the given {@code predicate}.
+     * @throws NullPointerException if {@code predicate} is null.
+     */
+    void updateFilteredTaskList(Predicate<Task> predicate);
+
+    /** Returns an unmodifiable view of the filtered person list */
+    ObservableList<Task> getFilteredTaskList();
+
+    //=========== Person oriented methods (OLD CODE) =============================================================
+
```

Comment:
TODO: extract this out of `Model` and put it into `TaskList`. Model is a high-level component - passing `Model` to the `execute` method creates a strong cyclic dependency.

Reply:
So u are saying we should not pass `Model `to the `execute `method?